content/docs/guides/crosswalk/kubernetes/introduction/01-architecture.md

```diff
@@ -0,0 +1,127 @@
+  can't accidentally destroy the primary datastore.
+* **Independently managable.** For example, the identity stack (containing, e.g., GCP IAM) can
+  expose roles and credentials; the permissions of a specific role can be changed without making
+  changes to (say) the application layer.
+
+We advocate splitting infrastructure up into (roughly) 6 stacks of resources. As we will see, this is
+particularly easy with Pulumi, as the notion of a [Pulumi
+stack]({{< relref "/docs/intro/concepts/organizing-stacks-projects.md" >}}) was specifically designed for
+this use case.
+
+<center><img src="/images/docs/quickstart/kubernetes/cake.svg" width="670"></center>
+
+## 1. Identity
+
+Identities and role definitions for organizations and CI/CD are required before anyone can provision
+anything. This is a requirement for every production Kubernetes deployment.
+
+One side-effect of isolating resources into loosely-coupled stacks is that we
+have the opportunity to grant **minimal permissions** based on need. (See also
+[principle of least
+privilege](https://en.m.wikipedia.org/wiki/Principleofleastprivilege)).
+
+Typically the identity stack looks like:
+
+1. **AWS/Azure/GCP identities and roles for the team.** (e.g., [AWS IAM][aws-iam], [GCP
+   IAM][gcp-iam], [Azure AD][azure-ad].) For example, the databases team typically gets only
+   administrative permissions for the datastores, while an app team might only get cluster
+   developer permissions.
+1. **AWS/Azure/GCP service accounts for CI/CD.** While IAM roles and Active Directory accounts
+   describe identity of users, service accounts grant an identity for workloads, e.g. Storage
+   CI/CD. The identity stack provisions these as well, with similarly minimal permissions. For
+   example, the storage infrastructure service account may only have administrative permissions in its given namespace.
+
+## 2. Managed Infrastructure
+
+Provisioning shared, managed infrastructure is required to configure the
+cluster. At minimum, this typically this includes networking infrastructure,
+and can often include a managed datastore (e.g., [Aurora][aurora],
+[Cloud SQL][cloud-sql], [CosmosDB][cosmos-db]) along with other cloud services such as VMs, registries, data pipelines, and data warehouses.
+
+## 3. Kubernetes Cluster 
+
+Provision the managed Kubernetes cluster on [EKS][eks], [GKE][gke], or [AKS][aks],
+and configure it with the desired settings and defaults.
+
+Typically this also involves provisioning Kubernetes cluster infrastructure
+such as `Namespaces`, `Roles` , `RoleBindings`, `Quotas` etc.
+
+## 4. Cluster Services
+
+With a running, vanilla cluster you should install any Kubernetes cluster-scoped
+services that will be shared by a subset or all cluster users.
+
+At a minimum, services include centralized cluster and app-based logging, and often
+include monitoring, policies, registries, and service meshes.
+
+## 5. App Services
```

Comment:
I was surprised that "data stores" didn't go in this section (e.g., databases, queues, topics, etc).

Reply:
Good point. Originally these were in 2. Managed Infrastructure, but i've rephrased that section to be about storage backends, and moved the datastores into the 5. App services as it seems fitting to go here instead.

Done